.gitlab-ci.yml

```diff
@@ -1352,3 +1365,19 @@ workflow:
 .on_fakeintake_changes_on_main:
   - <<: *on_fakeintake_changes
     if: $CI_COMMIT_BRANCH == "main"
+
+.fast_on_dev_branch:
+  - <<: *if_main_branch
+    variables:
+      FAST_TESTS: "false"
+  - <<: *if_release_branch
+    variables:
+      FAST_TESTS: "false"
+  - <<: *if_triggered_pipeline
+    variables:
+      FAST_TESTS: "false"
+  - <<: *if_run_all_unit_tests
+    variables:
+      FAST_TESTS: "false"
+  - variables:
```

Comment:
I cannot really see another way to use the fast unit tests on PR branches only, since PR branches can have an arbitrary name.
I could have the first condition being something like:
```
- if: not_main && not_release && not triggered_pipeline && run_all_unit_tests
   variables:
       FAST_TESTS: true
- variables:
       FAST_TESTS: false
```

But I feel like it make the conditions more complex

Reply:
Updated after our discussion